### DISCUSSION TITLE
db_query error:pg: division by zery under Dashboards-Statistics-Charges

### DISCUSSION THREAD

**Discussion by hofmarkus** [tillsteinbach/VWsFriend#153] [Q&A]
I have been using VWsFriends with my VW E-UP for about a week now. I am very satisfied with the data and the evaluation.

I have one question:
When I now go to "Dashboards/Statistics" I see the charges from the month. Until 3-4 days ago, I could see the approximate charge of the individual days under "Charges". Currently, however, I only see the table displayed here in the window and "Charges" now says 9 Charges and when I click there I only see the TOP 10 Charges at the right and under CHARGES I see a error "!" with "db query error: pq: division by zero". What could be the error and how could I correct this.

Have one more note. I also used VWsFriend together with ABRP and when starting the car this works relatively well, but unfortunately the data is hardly or not at all updated while driving and so I still have the somewhat inaccurate numbers from ABRP. When the car stops again, an update comes in after a while and when I select a new route I can start again with the current data. Can I actively improve this?
Thank you.

**Comment by tillsteinbach** [comment]
I will have a look what might be the problem with the charges dashboard. I will come back to you with more questions soon.

Regarding ABRP:
The problem is that the older platforms (before ID) do not transmit the data when the ignition is on. I have the same situation with my Passat GTE. The ID.3 of my wife sends its data all the time. That makes it much more useful for ABRP when traveling.

**Reply by hofmarkus** [reply]
Thank you very much for your answer. I hope there is a solution to the Charges Dashboard. The information, even if only approximate values, was quite good.

ABRP:
If you would like to have more information, please tell me the commands I should execute to get the information.

That could be the problem. It is only updated every few minutes and you can't influence that.

**Reply by tillsteinbach** [reply]
I think I found the problem in the dashboard. It probably happens when there are two charging sessions with the same kilometers. That would mean that the consumption is infinite. I fixed that. Will be resolved in the next release coming soon.

**Reply by tillsteinbach** [reply]
The division error should be gone after update to the latest release. If not please drop me a comment.

**Reply by hofmarkus** [reply]
I have just updated the software and tested it again and can happily confirm that the problem has now been solved.
Thank you very much, everything is now working fine (division error is fixed).

**Comment by hofmarkus** [comment]
This is good news. 
I will test the new version soon. 

I wanted to ask if it is possible to connect not only with "We Connect ID" to read out the data, but also with "We Connect". With the VW E-UP, you could read out a lot more interesting data here. 
Data in "We Connect" would be:
Driving distance
Driving time
Average speed
Average consumption
Average Recuperation
Average internal consumption
Programming of charges + departure times + chargelimits
Unfortunately, some of this data cannot be read out with We Connect ID (perhaps with another car?).

Thanks again.

**Reply by tillsteinbach** [reply]
Currently I don’t plan that as I expect that the ID API will get the same features soon. It is too much overhead to maintain both APIs. As soon as the trips make it in the API I will add those.

**Reply by hofmarkus** [reply]
Yes, I understand the decision. Let's hope in the future more data will be available via We Connect ID.
Thank you very much.
